iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/utils/executor/batch/utils/AlignedSeriesBatchCompactionUtils.java

```diff
@@ -0,0 +1,175 @@
+      if (!isLargeDataType(schema.getType())) {
+        continue;
+      }
+      if (compactedMeasurements.contains(schema.getMeasurementId())) {
+        continue;
+      }
+      compactedMeasurements.add(schema.getMeasurementId());
+      selectedColumnBatch.add(schema);
+      if (selectedColumnBatch.size() >= batchSize) {
+        return selectedColumnBatch;
+      }
+      if (compactedMeasurements.size() == schemaList.size()) {
+        return selectedColumnBatch;
+      }
+    }
+    for (IMeasurementSchema schema : schemaList) {
+      if (compactedMeasurements.contains(schema.getMeasurementId())) {
+        continue;
+      }
+      selectedColumnBatch.add(schema);
+      compactedMeasurements.add(schema.getMeasurementId());
+      if (selectedColumnBatch.size() >= batchSize) {
+        break;
+      }
+      if (compactedMeasurements.size() == schemaList.size()) {
+        break;
+      }
+    }
+    return selectedColumnBatch;
+  }
+
+  private static boolean isLargeDataType(TSDataType dataType) {
+    return dataType.equals(TSDataType.BLOB)
+        || dataType.equals(TSDataType.TEXT)
+        || dataType.equals(TSDataType.STRING);
+  }
+
+  public static void markAlignedChunkHasDeletion(
+      LinkedList<Pair<TsFileSequenceReader, List<AlignedChunkMetadata>>>
+          readerAndChunkMetadataList) {
+    for (Pair<TsFileSequenceReader, List<AlignedChunkMetadata>> pair : readerAndChunkMetadataList) {
+      List<AlignedChunkMetadata> alignedChunkMetadataList = pair.getRight();
+      markAlignedChunkHasDeletion(alignedChunkMetadataList);
+    }
+  }
+
+  public static void markAlignedChunkHasDeletion(
+      List<AlignedChunkMetadata> alignedChunkMetadataList) {
+    for (AlignedChunkMetadata alignedChunkMetadata : alignedChunkMetadataList) {
+      IChunkMetadata timeChunkMetadata = alignedChunkMetadata.getTimeChunkMetadata();
+      for (IChunkMetadata iChunkMetadata : alignedChunkMetadata.getValueChunkMetadataList()) {
+        if (iChunkMetadata != null && iChunkMetadata.isModified()) {
+          timeChunkMetadata.setModified(true);
+          break;
+        }
+      }
+    }
+  }
+
+  public static AlignedChunkMetadata filterAlignedChunkMetadata(
+      AlignedChunkMetadata alignedChunkMetadata, List<String> selectedMeasurements) {
+    List<IChunkMetadata> valueChunkMetadataList =
+        Arrays.asList(new IChunkMetadata[selectedMeasurements.size()]);
+
+    Map<String, Integer> measurementIndex = new HashMap<>();
+    for (int i = 0; i < selectedMeasurements.size(); i++) {
+      measurementIndex.put(selectedMeasurements.get(i), i);
+    }
+
+    for (IChunkMetadata chunkMetadata : alignedChunkMetadata.getValueChunkMetadataList()) {
+      if (chunkMetadata == null) {
+        continue;
+      }
+      if (measurementIndex.containsKey(chunkMetadata.getMeasurementUid())) {
+        valueChunkMetadataList.set(
+            measurementIndex.get(chunkMetadata.getMeasurementUid()), chunkMetadata);
+      }
```

Comment:
The measurementIndex is not modified here, but the valueChunkMetadataList is modified. Using computeIfPresent may seem a bit strange.

Reply:
Not modifying measurementIndex , just to avoid accessing the map twice.
```suggestion
      chunkMetadataMap.computeIfPresent(chunkMetadata.getMeasurementUid(),
        (k, v) -> {
          valueChunkMetadataList.add(v);
          return v;
        }
    );
```

Reply:
Not modifying measurementIndex , just to avoid accessing the map twice.
```suggestion
      chunkMetadataMap.computeIfPresent(chunkMetadata.getMeasurementUid(),
        (k, v) -> {
          valueChunkMetadataList.add(v);
          return v;
        }
    );
```